### ISSUE TITLE
Rare coverage ID sync failures

### ISSUE THREAD

**Issue by fmeum** [CodeIntelligenceTesting/jazzer#86]
As reported by ClusterFuzz, coverage IDs can become desynchronized between child processes in fork mode. I attached a log obtained from the `ValidJsonFuzzer` fuzzer of the `json-sanitizer` project.

@sebastianpoeplau Could you review https://github.com/CodeIntelligenceTesting/jazzer/blob/main/agent/src/main/java/com/code_intelligence/jazzer/agent/CoverageIdStrategy.kt again for potential synchronization issues? Any idea on how to improve logging of these cases further would also be helpful.

```
[Environment] ASAN_OPTIONS=alloc_dealloc_mismatch=0:allocator_may_return_null=1:allocator_release_to_os_interval_ms=500:allow_user_segv_handler=0:check_malloc_usable_size=0:detect_leaks=1:detect_odr_violation=0:detect_stack_use_after_return=1:fast_unwind_on_fatal=0:handle_abort=2:handle_segv=2:handle_sigbus=2:handle_sigfpe=2:handle_sigill=2:max_uar_stack_size_log=16:print_scariness=1:print_summary=1:print_suppressions=0:quarantine_size_mb=64:redzone=128:strict_memcmp=1:strip_path_prefix=/workspace/:symbolize=0:use_sigaltstack=1
+----------------------------------------Release Build Stacktrace----------------------------------------+
INFO: Synchronizing coverage IDs in /tmp/jazzer-rjzvredadvdehcizmmuolhlnhchnqfxs
INFO: Loaded 8649 no-throw method signatures
INFO: Instrumented ValidJsonFuzzer (took 156 ms, size +37%)
INFO: Instrumented com.google.gson.Gson (took 85 ms, size +32%)
INFO: Instrumented com.google.gson.stream.MalformedJsonException (took 3 ms, size +37%)
INFO: Instrumented com.google.gson.JsonIOException (took 3 ms, size +46%)
INFO: Instrumented com.google.gson.JsonParseException (took 3 ms, size +36%)
INFO: Instrumented com.google.gson.JsonSyntaxException (took 3 ms, size +45%)
INFO: New number of inline 8-bit counters: 1024
INFO: Instrumented com.google.gson.stream.JsonReader (took 75 ms, size +85%)
INFO: New number of inline 8-bit counters: 2048
INFO: Instrumented com.google.gson.internal.bind.JsonTreeReader (took 23 ms, size +63%)
INFO: Instrumented com.google.gson.TypeAdapter (took 5 ms, size +25%)
INFO: Instrumented com.google.gson.Gson$3 (took 8 ms, size +29%)
INFO: Instrumented com.google.gson.Gson$1 (took 4 ms, size +29%)
INFO: Instrumented com.google.gson.Gson$2 (took 4 ms, size +29%)
INFO: Instrumented com.google.gson.TypeAdapterFactory (took 2 ms, size +0%)
INFO: Instrumented com.google.gson.JsonElement (took 7 ms, size +32%)
INFO: Instrumented com.google.gson.JsonNull (took 5 ms, size +66%)
INFO: Instrumented com.google.gson.stream.JsonWriter (took 35 ms, size +67%)
INFO: Instrumented com.google.gson.internal.bind.JsonTreeWriter (took 10 ms, size +53%)
INFO: Instrumented com.google.gson.FieldNamingStrategy (took 2 ms, size +0%)
INFO: Instrumented com.google.gson.Gson$FutureTypeAdapter (took 4 ms, size +26%)
INFO: Instrumented com.google.gson.reflect.TypeToken (took 11 ms, size +40%)
INFO: Instrumented com.google.gson.internal.$Gson$Preconditions (took 3 ms, size +50%)
INFO: Instrumented com.google.gson.internal.$Gson$Types (took 30 ms, size +59%)
INFO: Instrumented com.google.gson.internal.Excluder (took 18 ms, size +61%)
INFO: Instrumented com.google.gson.internal.Excluder$1 (took 11 ms, size +31%)
INFO: Instrumented com.google.gson.FieldNamingPolicy (took 8 ms, size +40%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$1 (took 5 ms, size +37%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$2 (took 5 ms, size +34%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$3 (took 7 ms, size +33%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$4 (took 9 ms, size +28%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$5 (took 2 ms, size +28%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$6 (took 4 ms, size +28%)
INFO: Instrumented com.google.gson.LongSerializationPolicy (took 22 ms, size +30%)
INFO: Instrumented com.google.gson.LongSerializationPolicy$1 (took 5 ms, size +39%)
INFO: Instrumented com.google.gson.LongSerializationPolicy$2 (took 2 ms, size +33%)
INFO: New number of inline 8-bit counters: 4096
INFO: Instrumented com.google.gson.JsonPrimitive (took 10 ms, size +81%)
INFO: Instrumented com.google.gson.internal.ConstructorConstructor (took 6 ms, size +33%)
INFO: Instrumented com.google.gson.internal.ObjectConstructor (took 3 ms, size +0%)
INFO: Instrumented com.google.gson.internal.reflect.ReflectionAccessor (took 5 ms, size +57%)
INFO: Instrumented com.google.gson.internal.reflect.PreJava9ReflectionAccessor (took 2 ms, size +44%)
INFO: Instrumented com.google.gson.internal.reflect.UnsafeReflectionAccessor (took 11 ms, size +45%)
INFO: Instrumented com.google.gson.internal.JavaVersion (took 7 ms, size +61%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters (took 11 ms, size +19%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$3 (took 9 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$4 (took 3 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$5 (took 2 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$6 (took 2 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$7 (took 2 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$11 (took 2 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$12 (took 2 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$13 (took 2 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$14 (took 3 ms, size +26%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$15 (took 3 ms, size +34%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$16 (took 2 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$17 (took 2 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$18 (took 2 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$19 (took 3 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$20 (took 3 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$21 (took 3 ms, size +57%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$22 (took 3 ms, size +56%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$23 (took 3 ms, size +32%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$24 (took 2 ms, size +33%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$27 (took 5 ms, size +75%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$28 (took 4 ms, size +49%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$29 (took 5 ms, size +50%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$1 (took 3 ms, size +23%)
INFO: Instrumented com.google.gson.TypeAdapter$1 (took 3 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$32 (took 2 ms, size +19%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$2 (took 4 ms, size +44%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$33 (took 3 ms, size +20%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$8 (took 3 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$9 (took 2 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$10 (took 4 ms, size +37%)
INFO: Instrumented com.google.gson.internal.LazilyParsedNumber (took 4 ms, size +69%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$35 (took 6 ms, size +24%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$35$1 (took 4 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$25 (took 3 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$26 (took 4 ms, size +22%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$26$1 (took 3 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$34 (took 4 ms, size +20%)
INFO: Instrumented com.google.gson.JsonArray (took 6 ms, size +59%)
INFO: Instrumented com.google.gson.JsonObject (took 4 ms, size +44%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$30 (took 2 ms, size +39%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$EnumTypeAdapter (took 4 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.ObjectTypeAdapter (took 4 ms, size +41%)
INFO: Instrumented com.google.gson.internal.bind.ObjectTypeAdapter$1 (took 4 ms, size +27%)
INFO: Instrumented com.google.gson.Gson$4 (took 14 ms, size +25%)
INFO: Instrumented com.google.gson.Gson$5 (took 4 ms, size +36%)
INFO: Instrumented com.google.gson.internal.bind.DateTypeAdapter (took 9 ms, size +35%)
INFO: Instrumented com.google.gson.internal.bind.DateTypeAdapter$1 (took 3 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TimeTypeAdapter (took 5 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TimeTypeAdapter$1 (took 3 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.SqlDateTypeAdapter (took 3 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.SqlDateTypeAdapter$1 (took 3 ms, size +28%)
INFO: Instrumented com.google.gson.internal.bind.ArrayTypeAdapter (took 7 ms, size +34%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper (took 4 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.ArrayTypeAdapter$1 (took 5 ms, size +33%)
INFO: Instrumented com.google.gson.internal.bind.CollectionTypeAdapterFactory (took 3 ms, size +21%)
INFO: Instrumented com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter (took 4 ms, size +28%)
INFO: Instrumented com.google.gson.internal.bind.MapTypeAdapterFactory (took 3 ms, size +23%)
INFO: Instrumented com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter (took 6 ms, size +39%)
INFO: Instrumented com.google.gson.internal.bind.JsonAdapterAnnotationTypeAdapterFactory (took 4 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TreeTypeAdapter (took 9 ms, size +22%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory (took 9 ms, size +28%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$BoundField (took 3 ms, size +21%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1 (took 4 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter (took 4 ms, size +32%)
INFO: libFuzzer ignores flags that start with '--'
Dictionary: 44 entries
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 3065147706
INFO: Loaded 4 modules   (4096 inline 8-bit counters): 512 [0x7fbb90ef7010, 0x7fbb90ef7210), 512 [0x7fbb90ef7210, 0x7fbb90ef7410), 1024 [0x7fbb90ef7410, 0x7fbb90ef7810), 2048 [0x7fbb90ef7810, 0x7fbb90ef8010),
INFO: Loaded 4 PC tables (4096 PCs): 512 [0x7fbb62fff010,0x7fbb63001010), 512 [0x7fbb63001010,0x7fbb63003010), 1024 [0x7fbb63003010,0x7fbb63007010), 2048 [0x7fbb63007010,0x7fbb6300f010),
INFO: -fork=2: fuzzing in separate process(s)
INFO: -fork=2: 10 seed inputs, starting to fuzz in /tmp/libFuzzerTemp.FuzzWithFork15947.dir
#545: cov: 2054 ft: 2054 corp: 10 exec/s 272 oom/timeout/crash: 0/0/0 time: 6s job: 1 dft_time: 0
#1464: cov: 2730 ft: 3585 corp: 162 exec/s 306 oom/timeout/crash: 0/0/0 time: 10s job: 2 dft_time: 0
#1850: cov: 2845 ft: 4170 corp: 277 exec/s 386 oom/timeout/crash: 0/0/0 time: 19s job: 3 dft_time: 0
INFO: log from the inner process:
INFO: Synchronizing coverage IDs in /tmp/jazzer-rjzvredadvdehcizmmuolhlnhchnqfxs
INFO: Loaded 8649 no-throw method signatures
INFO: Instrumented ValidJsonFuzzer (took 325 ms, size +37%)
INFO: Instrumented com.google.gson.Gson (took 128 ms, size +32%)
INFO: Instrumented com.google.gson.stream.MalformedJsonException (took 13 ms, size +37%)
INFO: Instrumented com.google.gson.JsonIOException (took 28 ms, size +46%)
INFO: Instrumented com.google.gson.JsonParseException (took 8 ms, size +36%)
INFO: Instrumented com.google.gson.JsonSyntaxException (took 41 ms, size +45%)
INFO: New number of inline 8-bit counters: 1024
INFO: Instrumented com.google.gson.stream.JsonReader (took 194 ms, size +85%)
INFO: New number of inline 8-bit counters: 2048
INFO: Instrumented com.google.gson.internal.bind.JsonTreeReader (took 71 ms, size +63%)
INFO: Instrumented com.google.gson.TypeAdapter (took 32 ms, size +25%)
INFO: Instrumented com.google.gson.Gson$3 (took 18 ms, size +29%)
INFO: Instrumented com.google.gson.Gson$1 (took 13 ms, size +29%)
INFO: Instrumented com.google.gson.Gson$2 (took 18 ms, size +29%)
INFO: Instrumented com.google.gson.TypeAdapterFactory (took 11 ms, size +0%)
INFO: Instrumented com.google.gson.JsonElement (took 26 ms, size +32%)
INFO: Instrumented com.google.gson.JsonNull (took 12 ms, size +66%)
INFO: Instrumented com.google.gson.stream.JsonWriter (took 67 ms, size +67%)
INFO: Instrumented com.google.gson.internal.bind.JsonTreeWriter (took 34 ms, size +53%)
INFO: Instrumented com.google.gson.FieldNamingStrategy (took 5 ms, size +0%)
INFO: Instrumented com.google.gson.Gson$FutureTypeAdapter (took 8 ms, size +26%)
INFO: Instrumented com.google.gson.reflect.TypeToken (took 29 ms, size +40%)
INFO: Instrumented com.google.gson.internal.$Gson$Preconditions (took 3 ms, size +50%)
INFO: Instrumented com.google.gson.internal.$Gson$Types (took 44 ms, size +59%)
INFO: Instrumented com.google.gson.internal.Excluder (took 18 ms, size +61%)
INFO: Instrumented com.google.gson.internal.Excluder$1 (took 10 ms, size +31%)
INFO: Instrumented com.google.gson.FieldNamingPolicy (took 4 ms, size +40%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$1 (took 7 ms, size +37%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$2 (took 5 ms, size +34%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$3 (took 2 ms, size +33%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$4 (took 2 ms, size +28%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$5 (took 9 ms, size +28%)
INFO: Instrumented com.google.gson.FieldNamingPolicy$6 (took 5 ms, size +28%)
INFO: Instrumented com.google.gson.LongSerializationPolicy (took 7 ms, size +30%)
INFO: Instrumented com.google.gson.LongSerializationPolicy$1 (took 2 ms, size +39%)
INFO: Instrumented com.google.gson.LongSerializationPolicy$2 (took 4 ms, size +33%)
INFO: New number of inline 8-bit counters: 4096
INFO: Instrumented com.google.gson.JsonPrimitive (took 16 ms, size +81%)
INFO: Instrumented com.google.gson.internal.ConstructorConstructor (took 16 ms, size +33%)
INFO: Instrumented com.google.gson.internal.ObjectConstructor (took 2 ms, size +0%)
INFO: Instrumented com.google.gson.internal.reflect.ReflectionAccessor (took 3 ms, size +57%)
INFO: Instrumented com.google.gson.internal.reflect.PreJava9ReflectionAccessor (took 13 ms, size +44%)
INFO: Instrumented com.google.gson.internal.reflect.UnsafeReflectionAccessor (took 15 ms, size +45%)
INFO: Instrumented com.google.gson.internal.JavaVersion (took 9 ms, size +61%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters (took 12 ms, size +19%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$3 (took 3 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$4 (took 2 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$5 (took 10 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$6 (took 6 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$7 (took 2 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$11 (took 6 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$12 (took 2 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$13 (took 2 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$14 (took 2 ms, size +26%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$15 (took 6 ms, size +34%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$16 (took 9 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$17 (took 2 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$18 (took 4 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$19 (took 6 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$20 (took 6 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$21 (took 7 ms, size +57%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$22 (took 4 ms, size +56%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$23 (took 5 ms, size +32%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$24 (took 6 ms, size +33%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$27 (took 11 ms, size +75%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$28 (took 3 ms, size +49%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$29 (took 10 ms, size +50%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$1 (took 7 ms, size +23%)
INFO: Instrumented com.google.gson.TypeAdapter$1 (took 14 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$32 (took 2 ms, size +19%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$2 (took 10 ms, size +44%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$33 (took 3 ms, size +20%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$8 (took 6 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$9 (took 2 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$10 (took 3 ms, size +37%)
INFO: Instrumented com.google.gson.internal.LazilyParsedNumber (took 10 ms, size +69%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$35 (took 10 ms, size +24%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$35$1 (took 3 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$25 (took 4 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$26 (took 2 ms, size +22%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$26$1 (took 2 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$34 (took 3 ms, size +20%)
INFO: Instrumented com.google.gson.JsonArray (took 16 ms, size +59%)
INFO: Instrumented com.google.gson.JsonObject (took 12 ms, size +44%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$30 (took 2 ms, size +39%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$EnumTypeAdapter (took 19 ms, size +30%)
INFO: Instrumented com.google.gson.internal.bind.ObjectTypeAdapter (took 5 ms, size +41%)
INFO: Instrumented com.google.gson.internal.bind.ObjectTypeAdapter$1 (took 9 ms, size +27%)
INFO: Instrumented com.google.gson.Gson$4 (took 13 ms, size +25%)
INFO: Instrumented com.google.gson.Gson$5 (took 11 ms, size +36%)
INFO: Instrumented com.google.gson.internal.bind.DateTypeAdapter (took 16 ms, size +35%)
INFO: Instrumented com.google.gson.internal.bind.DateTypeAdapter$1 (took 5 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.TimeTypeAdapter (took 10 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.TimeTypeAdapter$1 (took 2 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.SqlDateTypeAdapter (took 15 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.SqlDateTypeAdapter$1 (took 13 ms, size +28%)
INFO: Instrumented com.google.gson.internal.bind.ArrayTypeAdapter (took 10 ms, size +34%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper (took 9 ms, size +31%)
INFO: Instrumented com.google.gson.internal.bind.ArrayTypeAdapter$1 (took 7 ms, size +33%)
INFO: Instrumented com.google.gson.internal.bind.CollectionTypeAdapterFactory (took 5 ms, size +21%)
INFO: Instrumented com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter (took 9 ms, size +28%)
INFO: Instrumented com.google.gson.internal.bind.MapTypeAdapterFactory (took 3 ms, size +23%)
INFO: Instrumented com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter (took 13 ms, size +39%)
INFO: Instrumented com.google.gson.internal.bind.JsonAdapterAnnotationTypeAdapterFactory (took 9 ms, size +27%)
INFO: Instrumented com.google.gson.internal.bind.TreeTypeAdapter (took 5 ms, size +22%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory (took 7 ms, size +28%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$BoundField (took 15 ms, size +21%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1 (took 4 ms, size +29%)
INFO: Instrumented com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter (took 17 ms, size +32%)
INFO: libFuzzer ignores flags that start with '--'
Dictionary: 44 entries
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 3078689727
INFO: Loaded 4 modules   (4096 inline 8-bit counters): 512 [0x7ff492ecc010, 0x7ff492ecc210), 512 [0x7ff492ecc210, 0x7ff492ecc410), 1024 [0x7ff492ecc410, 0x7ff492ecc810), 2048 [0x7ff492ecc810, 0x7ff492ecd010),
INFO: Loaded 4 PC tables (4096 PCs): 512 [0x7ff491aca010,0x7ff491acc010), 512 [0x7ff491acc010,0x7ff491ace010), 1024 [0x7ff491ace010,0x7ff491ad2010), 2048 [0x7ff491ad2010,0x7ff491ada010),
INFO:        0 files found in /tmp/libFuzzerTemp.FuzzWithFork15947.dir/C3
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: New number of inline 8-bit counters: 8192
INFO: Instrumented com.google.json.JsonSanitizer (took 102 ms, size +119%)
INFO: Instrumented com.google.json.JsonSanitizer$UnbracketedComma (took 10 ms, size +39%)
INFO: Instrumented com.google.json.JsonSanitizer$State (took 3 ms, size +39%)
INFO: Instrumented com.google.json.JsonSanitizer$1 (took 2 ms, size +74%)
INFO: Instrumented com.google.gson.internal.JsonReaderInternalAccess (took 2 ms, size +44%)
INFO: Instrumented com.google.gson.stream.JsonReader$1 (took 2 ms, size +53%)
INFO: Instrumented com.google.gson.stream.JsonToken (took 3 ms, size +40%)
INFO: Instrumented com.google.gson.internal.bind.TypeAdapters$36 (took 9 ms, size +81%)
INFO: Instrumented com.google.gson.internal.Primitives (took 20 ms, size +80%)
INFO: seed corpus: files: 12 min: 17b max: 2048b total: 6632b rss: 155Mb
#13 INITED cov: 512 ft: 1781 corp: 12/6632b exec/s: 0 rss: 156Mb
#15 REDUCE cov: 512 ft: 1781 corp: 12/6625b lim: 2048 exec/s: 0 rss: 156Mb L: 53/2048 MS: 2 ChangeBinInt-EraseBytes-
#16 NEW    cov: 512 ft: 1795 corp: 13/6695b lim: 2048 exec/s: 0 rss: 156Mb L: 70/2048 MS: 1 InsertByte-
#17 NEW    cov: 512 ft: 1801 corp: 14/8743b lim: 2048 exec/s: 0 rss: 156Mb L: 2048/2048 MS: 1 ChangeBinInt-
#27 NEW    cov: 512 ft: 1812 corp: 15/8869b lim: 2048 exec/s: 0 rss: 156Mb L: 126/2048 MS: 5 ChangeByte-ChangeASCIIInt-ChangeBit-ChangeBinInt-InsertRepeatedBytes-
#31 NEW    cov: 512 ft: 1830 corp: 16/9119b lim: 2048 exec/s: 0 rss: 156Mb L: 250/2048 MS: 4 ShuffleBytes-ChangeBinInt-CrossOver-CrossOver-
#32 NEW    cov: 512 ft: 1833 corp: 17/11167b lim: 2048 exec/s: 0 rss: 156Mb L: 2048/2048 MS: 1 ChangeBit-
#34 NEW    cov: 512 ft: 1839 corp: 18/11357b lim: 2048 exec/s: 0 rss: 156Mb L: 190/2048 MS: 2 ChangeBit-CMP- DE: "\x00\x1c"-
#35 NEW    cov: 512 ft: 1880 corp: 19/13405b lim: 2048 exec/s: 0 rss: 156Mb L: 2048/2048 MS: 1 CMP- DE: "\x17\x01\x00\x00"-
#36 NEW    cov: 522 ft: 1938 corp: 20/13625b lim: 2048 exec/s: 0 rss: 156Mb L: 220/2048 MS: 1 CrossOver-
#37 NEW    cov: 522 ft: 1939 corp: 21/13642b lim: 2048 exec/s: 0 rss: 156Mb L: 17/2048 MS: 1 ChangeASCIIInt-
#44 NEW    cov: 522 ft: 1964 corp: 22/13663b lim: 2048 exec/s: 0 rss: 156Mb L: 21/2048 MS: 2 PersAutoDict-ChangeBinInt- DE: "\x17\x01\x00\x00"-
#47 NEW    cov: 522 ft: 1969 corp: 23/15711b lim: 2048 exec/s: 0 rss: 156Mb L: 2048/2048 MS: 3 ChangeASCIIInt-ChangeBit-ChangeByte-
#48 NEW    cov: 522 ft: 1986 corp: 24/15899b lim: 2048 exec/s: 0 rss: 156Mb L: 188/2048 MS: 1 InsertRepeatedBytes-
#49 NEW    cov: 522 ft: 2007 corp: 25/15959b lim: 2048 exec/s: 0 rss: 156Mb L: 60/2048 MS: 1 CrossOver-
#50 NEW    cov: 522 ft: 2008 corp: 26/15976b lim: 2048 exec/s: 0 rss: 156Mb L: 17/2048 MS: 1 ChangeBit-
#52 REDUCE cov: 522 ft: 2008 corp: 26/15954b lim: 2048 exec/s: 0 rss: 156Mb L: 31/2048 MS: 2 CMP-EraseBytes- DE: "\x00\x00\x00\x00\x00\x00\x00\x00"-
#53 NEW    cov: 522 ft: 2015 corp: 27/16146b lim: 2048 exec/s: 0 rss: 156Mb L: 192/2048 MS: 1 PersAutoDict- DE: "\x00\x1c"-
#59 NEW    cov: 523 ft: 2034 corp: 28/16176b lim: 2048 exec/s: 0 rss: 156Mb L: 30/2048 MS: 1 CrossOver-
#62 NEW    cov: 523 ft: 2046 corp: 29/17Kb lim: 2048 exec/s: 0 rss: 156Mb L: 2048/2048 MS: 3 ChangeBinInt-ChangeByte-ChangeByte-
#71 NEW    cov: 523 ft: 2053 corp: 30/17Kb lim: 2048 exec/s: 0 rss: 156Mb L: 99/2048 MS: 4 CopyPart-InsertByte-InsertByte-InsertRepeatedBytes-
#75 NEW    cov: 523 ft: 2059 corp: 31/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 229/2048 MS: 4 ChangeBinInt-ManualDict-PersAutoDict-InsertRepeatedBytes- DE: ":{}"-"\x00\x1c"-
#76 REDUCE cov: 523 ft: 2059 corp: 31/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 114/2048 MS: 1 EraseBytes-
#78 NEW    cov: 523 ft: 2081 corp: 32/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 169/2048 MS: 2 InsertRepeatedBytes-CrossOver-
#81 NEW    cov: 523 ft: 2088 corp: 33/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 61/2048 MS: 3 ChangeBit-ChangeByte-CopyPart-
#82 NEW    cov: 523 ft: 2103 corp: 34/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 33/2048 MS: 1 CMP- DE: "\x01C"-
#88 NEW    cov: 523 ft: 2111 corp: 35/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 21/2048 MS: 1 ChangeBinInt-
#89 NEW    cov: 523 ft: 2112 corp: 36/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 17/2048 MS: 1 EraseBytes-
#91 NEW    cov: 523 ft: 2121 corp: 37/18Kb lim: 2048 exec/s: 0 rss: 156Mb L: 30/2048 MS: 2 ChangeByte-ShuffleBytes-
#93 NEW    cov: 523 ft: 2123 corp: 38/19Kb lim: 2048 exec/s: 0 rss: 156Mb L: 1053/2048 MS: 2 CopyPart-CrossOver-
#98 NEW    cov: 523 ft: 2126 corp: 39/20Kb lim: 2048 exec/s: 0 rss: 160Mb L: 1377/2048 MS: 5 ChangeByte-ChangeByte-CopyPart-ChangeByte-EraseBytes-
#101 NEW    cov: 523 ft: 2132 corp: 40/22Kb lim: 2048 exec/s: 0 rss: 161Mb L: 2048/2048 MS: 3 ShuffleBytes-ChangeBit-ChangeByte-
#108 NEW    cov: 523 ft: 2137 corp: 41/22Kb lim: 2048 exec/s: 0 rss: 162Mb L: 33/2048 MS: 2 PersAutoDict-InsertByte- DE: "\x01C"-
#109 NEW    cov: 524 ft: 2150 corp: 42/22Kb lim: 2048 exec/s: 0 rss: 162Mb L: 95/2048 MS: 1 EraseBytes-
#115 NEW    cov: 524 ft: 2152 corp: 43/22Kb lim: 2048 exec/s: 0 rss: 162Mb L: 34/2048 MS: 1 CMP- DE: "\x01\x00\x00\x00\x00\x00\x00\x00"-
#116 NEW    cov: 524 ft: 2156 corp: 44/24Kb lim: 2048 exec/s: 0 rss: 162Mb L: 1378/2048 MS: 1 InsertByte-
#117 NEW    cov: 524 ft: 2160 corp: 45/24Kb lim: 2048 exec/s: 0 rss: 162Mb L: 62/2048 MS: 1 InsertByte-
#119 NEW    cov: 526 ft: 2169 corp: 46/24Kb lim: 2048 exec/s: 0 rss: 162Mb L: 114/2048 MS: 2 InsertRepeatedBytes-ManualDict- DE: "{\"foo\":t"-
#120 NEW    cov: 526 ft: 2174 corp: 47/26Kb lim: 2048 exec/s: 0 rss: 162Mb L: 2048/2048 MS: 1 ChangeBit-
#124 NEW    cov: 526 ft: 2179 corp: 48/26Kb lim: 2048 exec/s: 0 rss: 162Mb L: 193/2048 MS: 4 ChangeBit-InsertByte-InsertRepeatedBytes-InsertRepeatedBytes-
#125 NEW    cov: 526 ft: 2187 corp: 49/28Kb lim: 2048 exec/s: 0 rss: 162Mb L: 2048/2048 MS: 1 CopyPart-
#126 NEW    cov: 530 ft: 2196 corp: 50/28Kb lim: 2048 exec/s: 0 rss: 162Mb L: 17/2048 MS: 1 ChangeBit-
#132 NEW    cov: 530 ft: 2197 corp: 51/28Kb lim: 2048 exec/s: 0 rss: 162Mb L: 56/2048 MS: 1 InsertRepeatedBytes-
#133 REDUCE cov: 530 ft: 2197 corp: 51/27Kb lim: 2048 exec/s: 0 rss: 162Mb L: 761/2048 MS: 1 CrossOver-
#136 NEW    cov: 530 ft: 2199 corp: 52/27Kb lim: 2048 exec/s: 0 rss: 162Mb L: 12/2048 MS: 3 ChangeBinInt-EraseBytes-ManualDict- DE: "[0]"-
#137 NEW    cov: 530 ft: 2205 corp: 53/29Kb lim: 2048 exec/s: 0 rss: 162Mb L: 1893/2048 MS: 1 ShuffleBytes-
#138 NEW    cov: 531 ft: 2211 corp: 54/29Kb lim: 2048 exec/s: 0 rss: 162Mb L: 142/2048 MS: 1 InsertRepeatedBytes-
#139 NEW    cov: 531 ft: 2214 corp: 55/31Kb lim: 2048 exec/s: 0 rss: 162Mb L: 2048/2048 MS: 1 ChangeBit-
#140 NEW    cov: 531 ft: 2217 corp: 56/33Kb lim: 2048 exec/s: 0 rss: 162Mb L: 2048/2048 MS: 1 ChangeBinInt-
#143 NEW    cov: 531 ft: 2239 corp: 57/33Kb lim: 2048 exec/s: 0 rss: 162Mb L: 179/2048 MS: 3 CrossOver-ChangeASCIIInt-CrossOver-
#148 NEW    cov: 531 ft: 2243 corp: 58/33Kb lim: 2048 exec/s: 0 rss: 162Mb L: 17/2048 MS: 5 ChangeBinInt-ChangeBinInt-PersAutoDict-PersAutoDict-ChangeByte- DE: "\x01\x00\x00\x00\x00\x00\x00\x00"-"\x00\x00\x00\x00\x00\x00\x00\x00"-
#152 NEW    cov: 531 ft: 2274 corp: 59/35Kb lim: 2048 exec/s: 0 rss: 162Mb L: 2048/2048 MS: 4 ChangeBit-ChangeASCIIInt-ChangeASCIIInt-PersAutoDict- DE: ":{}"-
#153 NEW    cov: 531 ft: 2276 corp: 60/35Kb lim: 2048 exec/s: 0 rss: 162Mb L: 85/2048 MS: 1 CopyPart-
#156 NEW    cov: 531 ft: 2278 corp: 61/35Kb lim: 2048 exec/s: 0 rss: 162Mb L: 33/2048 MS: 3 ChangeBit-ShuffleBytes-ShuffleBytes-
#158 NEW    cov: 531 ft: 2288 corp: 62/37Kb lim: 2048 exec/s: 0 rss: 162Mb L: 2048/2048 MS: 2 ChangeBit-CrossOver-
#159 NEW    cov: 531 ft: 2293 corp: 63/39Kb lim: 2048 exec/s: 0 rss: 162Mb L: 2048/2048 MS: 1 ChangeBit-
#160 NEW    cov: 533 ft: 2309 corp: 64/39Kb lim: 2048 exec/s: 0 rss: 162Mb L: 30/2048 MS: 1 PersAutoDict- DE: "\x01\x00\x00\x00\x00\x00\x00\x00"-
#161 NEW    cov: 533 ft: 2310 corp: 65/39Kb lim: 2048 exec/s: 0 rss: 162Mb L: 126/2048 MS: 1 CrossOver-
#162 NEW    cov: 533 ft: 2311 corp: 66/39Kb lim: 2048 exec/s: 0 rss: 162Mb L: 99/2048 MS: 1 ChangeByte-
#166 NEW    cov: 533 ft: 2313 corp: 67/39Kb lim: 2048 exec/s: 0 rss: 162Mb L: 24/2048 MS: 4 CopyPart-ShuffleBytes-ShuffleBytes-ChangeBinInt-
#177 NEW    cov: 533 ft: 2327 corp: 68/40Kb lim: 2048 exec/s: 0 rss: 162Mb L: 86/2048 MS: 1 InsertRepeatedBytes-
#186 NEW    cov: 533 ft: 2328 corp: 69/42Kb lim: 2048 exec/s: 0 rss: 163Mb L: 2048/2048 MS: 4 ChangeByte-ChangeBit-ShuffleBytes-ChangeBinInt-
#187 NEW    cov: 533 ft: 2329 corp: 70/42Kb lim: 2048 exec/s: 0 rss: 163Mb L: 62/2048 MS: 1 ChangeBinInt-
#189 NEW    cov: 533 ft: 2331 corp: 71/42Kb lim: 2048 exec/s: 0 rss: 163Mb L: 192/2048 MS: 2 ManualDict-ChangeBit- DE: "//"-
#195 NEW    cov: 533 ft: 2332 corp: 72/43Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1173/2048 MS: 1 EraseBytes-
#196 NEW    cov: 533 ft: 2335 corp: 73/43Kb lim: 2048 exec/s: 0 rss: 164Mb L: 61/2048 MS: 1 CopyPart-
#199 NEW    cov: 533 ft: 2338 corp: 74/43Kb lim: 2048 exec/s: 0 rss: 164Mb L: 200/2048 MS: 3 ChangeASCIIInt-InsertRepeatedBytes-ManualDict- DE: "\\uDBFF\\uDFFF"-
#200 NEW    cov: 552 ft: 2409 corp: 75/43Kb lim: 2048 exec/s: 0 rss: 164Mb L: 18/2048 MS: 1 InsertByte-
#202 NEW    cov: 552 ft: 2411 corp: 76/44Kb lim: 2048 exec/s: 0 rss: 164Mb L: 283/2048 MS: 2 ChangeByte-InsertRepeatedBytes-
#203 NEW    cov: 552 ft: 2412 corp: 77/44Kb lim: 2048 exec/s: 0 rss: 164Mb L: 229/2048 MS: 1 ChangeBit-
#219 NEW    cov: 552 ft: 2413 corp: 78/44Kb lim: 2048 exec/s: 0 rss: 164Mb L: 38/2048 MS: 1 PersAutoDict- DE: "\x17\x01\x00\x00"-
#223 NEW    cov: 552 ft: 2414 corp: 79/46Kb lim: 2048 exec/s: 0 rss: 164Mb L: 2048/2048 MS: 4 ChangeByte-ChangeASCIIInt-ChangeASCIIInt-ChangeByte-
#227 NEW    cov: 552 ft: 2415 corp: 80/48Kb lim: 2048 exec/s: 0 rss: 164Mb L: 2048/2048 MS: 4 ShuffleBytes-ChangeBinInt-ChangeASCIIInt-ShuffleBytes-
#228 NEW    cov: 552 ft: 2416 corp: 81/48Kb lim: 2048 exec/s: 0 rss: 164Mb L: 129/2048 MS: 1 ManualDict- DE: "\"\":"-
#229 NEW    cov: 552 ft: 2419 corp: 82/48Kb lim: 2048 exec/s: 0 rss: 164Mb L: 21/2048 MS: 1 ManualDict- DE: "7"-
#230 NEW    cov: 552 ft: 2421 corp: 83/48Kb lim: 2048 exec/s: 0 rss: 164Mb L: 38/2048 MS: 1 PersAutoDict- DE: "\x00\x00\x00\x00\x00\x00\x00\x00"-
#236 NEW    cov: 552 ft: 2424 corp: 84/50Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1893/2048 MS: 1 ChangeBinInt-
#237 REDUCE cov: 552 ft: 2424 corp: 84/50Kb lim: 2048 exec/s: 0 rss: 164Mb L: 152/2048 MS: 1 EraseBytes-
#243 REDUCE cov: 552 ft: 2424 corp: 84/50Kb lim: 2048 exec/s: 0 rss: 164Mb L: 80/2048 MS: 1 EraseBytes-
#245 NEW    cov: 552 ft: 2430 corp: 85/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1763/2048 MS: 2 PersAutoDict-EraseBytes- DE: "\x01C"-
#246 NEW    cov: 552 ft: 2438 corp: 86/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 12/2048 MS: 1 PersAutoDict- DE: "//"-
#247 NEW    cov: 552 ft: 2439 corp: 87/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 33/2048 MS: 1 ChangeByte-
#249 NEW    cov: 563 ft: 2475 corp: 88/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 31/2048 MS: 2 CopyPart-InsertByte-
#252 NEW    cov: 563 ft: 2476 corp: 89/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 38/2048 MS: 3 ChangeBit-CMP-ChangeBinInt- DE: "\x01\x00\x00\x00\x00\x00\x00\x00"-
#253 NEW    cov: 563 ft: 2477 corp: 90/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 190/2048 MS: 1 ChangeBinInt-
#254 REDUCE cov: 563 ft: 2477 corp: 90/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 34/2048 MS: 1 EraseBytes-
#257 NEW    cov: 563 ft: 2485 corp: 91/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 134/2048 MS: 3 ChangeBinInt-CopyPart-PersAutoDict- DE: "\x00\x00\x00\x00\x00\x00\x00\x00"-
#258 NEW    cov: 563 ft: 2486 corp: 92/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 60/2048 MS: 1 CrossOver-
#262 NEW    cov: 563 ft: 2488 corp: 93/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 129/2048 MS: 4 InsertRepeatedBytes-EraseBytes-ChangeASCIIInt-InsertRepeatedBytes-
#268 NEW    cov: 563 ft: 2495 corp: 94/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 223/2048 MS: 1 InsertRepeatedBytes-
#273 NEW    cov: 563 ft: 2500 corp: 95/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 30/2048 MS: 5 CopyPart-ShuffleBytes-CMP-CopyPart-InsertByte- DE: "\x01\x00\x00\x00\x00\x00\x00\x00"-
#275 NEW    cov: 563 ft: 2509 corp: 96/52Kb lim: 2048 exec/s: 0 rss: 164Mb L: 33/2048 MS: 2 ChangeBinInt-ChangeByte-
#276 NEW    cov: 563 ft: 2513 corp: 97/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 142/2048 MS: 1 ManualDict- DE: "\\"-
#277 REDUCE cov: 563 ft: 2515 corp: 98/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 60/2048 MS: 1 EraseBytes-
#286 NEW    cov: 563 ft: 2517 corp: 99/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 155/2048 MS: 4 InsertRepeatedBytes-CMP-ChangeASCIIInt-CrossOver- DE: "FFE7"-
#292 REDUCE cov: 563 ft: 2517 corp: 99/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 9/2048 MS: 1 EraseBytes-
#293 NEW    cov: 563 ft: 2522 corp: 100/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 33/2048 MS: 1 CopyPart-
#294 NEW    cov: 563 ft: 2524 corp: 101/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 145/2048 MS: 1 ManualDict- DE: "@id"-
#301 NEW    cov: 563 ft: 2526 corp: 102/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 63/2048 MS: 2 ManualDict-ManualDict- DE: "\\n"-":"-
#304 REDUCE cov: 563 ft: 2526 corp: 102/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 24/2048 MS: 3 ChangeBinInt-CopyPart-EraseBytes-
#308 NEW    cov: 564 ft: 2528 corp: 103/53Kb lim: 2048 exec/s: 0 rss: 164Mb L: 35/2048 MS: 4 InsertByte-ManualDict-InsertByte-ManualDict- DE: "[]"-"\\"-
#311 NEW    cov: 564 ft: 2548 corp: 104/55Kb lim: 2048 exec/s: 0 rss: 164Mb L: 2048/2048 MS: 3 ChangeByte-ChangeBinInt-ChangeByte-
#312 NEW    cov: 564 ft: 2576 corp: 105/57Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1601/2048 MS: 1 EraseBytes-
#315 REDUCE cov: 564 ft: 2576 corp: 105/57Kb lim: 2048 exec/s: 0 rss: 164Mb L: 2034/2048 MS: 3 ChangeBinInt-ChangeBit-EraseBytes-
#316 NEW    cov: 564 ft: 2581 corp: 106/58Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1773/2048 MS: 1 CrossOver-
#324 NEW    cov: 564 ft: 2590 corp: 107/58Kb lim: 2048 exec/s: 0 rss: 164Mb L: 18/2048 MS: 3 ChangeASCIIInt-ChangeByte-ChangeByte-
#328 NEW    cov: 564 ft: 2592 corp: 108/59Kb lim: 2048 exec/s: 0 rss: 164Mb L: 306/2048 MS: 4 ShuffleBytes-InsertByte-InsertRepeatedBytes-CopyPart-
#329 NEW    cov: 564 ft: 2593 corp: 109/59Kb lim: 2048 exec/s: 0 rss: 164Mb L: 191/2048 MS: 1 InsertByte-
#330 NEW    cov: 564 ft: 2594 corp: 110/60Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1173/2048 MS: 1 ShuffleBytes-
#333 NEW    cov: 564 ft: 2595 corp: 111/60Kb lim: 2048 exec/s: 0 rss: 164Mb L: 63/2048 MS: 3 ChangeByte-InsertByte-ChangeByte-
#336 NEW    cov: 564 ft: 2604 corp: 112/61Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1398/2048 MS: 3 ChangeBinInt-ChangeBinInt-EraseBytes-
#340 NEW    cov: 566 ft: 2607 corp: 113/61Kb lim: 2048 exec/s: 0 rss: 164Mb L: 36/2048 MS: 4 InsertByte-ChangeByte-ChangeASCIIInt-ChangeByte-
#342 NEW    cov: 566 ft: 2610 corp: 114/61Kb lim: 2048 exec/s: 0 rss: 164Mb L: 56/2048 MS: 2 ChangeByte-ChangeByte-
#350 NEW    cov: 566 ft: 2616 corp: 115/62Kb lim: 2048 exec/s: 0 rss: 164Mb L: 223/2048 MS: 3 ShuffleBytes-ChangeByte-PersAutoDict- DE: "\\n"-
#352 REDUCE cov: 566 ft: 2619 corp: 116/64Kb lim: 2048 exec/s: 0 rss: 164Mb L: 2034/2048 MS: 2 ShuffleBytes-PersAutoDict- DE: "FFE7"-
#353 REDUCE cov: 566 ft: 2622 corp: 117/64Kb lim: 2048 exec/s: 0 rss: 164Mb L: 180/2048 MS: 1 InsertRepeatedBytes-
#356 NEW    cov: 566 ft: 2623 corp: 118/65Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1173/2048 MS: 3 ChangeByte-ChangeByte-ChangeBinInt-
#360 REDUCE cov: 569 ft: 2631 corp: 119/65Kb lim: 2048 exec/s: 0 rss: 164Mb L: 156/2048 MS: 4 ChangeByte-ChangeByte-ChangeBinInt-ManualDict- DE: "\\x00"-
#361 NEW    cov: 569 ft: 2633 corp: 120/67Kb lim: 2048 exec/s: 0 rss: 164Mb L: 2048/2048 MS: 1 CopyPart-
#362 NEW    cov: 582 ft: 2663 corp: 121/68Kb lim: 2048 exec/s: 0 rss: 164Mb L: 1173/2048 MS: 1 ChangeByte-
#363 NEW    cov: 582 ft: 2665 corp: 122/68Kb lim: 2048 exec/s: 0 rss: 164Mb L: 61/2048 MS: 1 ChangeBinInt-
#364 NEW    cov: 582 ft: 2666 corp: 123/68Kb lim: 2048 exec/s: 0 rss: 164Mb L: 38/2048 MS: 1 CopyPart-
#366 NEW    cov: 582 ft: 2667 corp: 124/70Kb lim: 2048 exec/s: 366 rss: 164Mb L: 2048/2048 MS: 2 CopyPart-ChangeBit-
#367 NEW    cov: 582 ft: 2668 corp: 125/72Kb lim: 2048 exec/s: 367 rss: 164Mb L: 2048/2048 MS: 1 ChangeBit-
#369 NEW    cov: 586 ft: 2676 corp: 126/73Kb lim: 2048 exec/s: 369 rss: 164Mb L: 296/2048 MS: 2 PersAutoDict-CrossOver- DE: ":"-
#370 NEW    cov: 586 ft: 2678 corp: 127/74Kb lim: 2048 exec/s: 370 rss: 164Mb L: 1775/2048 MS: 1 ManualDict- DE: "[]"-
#373 NEW    cov: 586 ft: 2679 corp: 128/74Kb lim: 2048 exec/s: 373 rss: 164Mb L: 54/2048 MS: 3 CopyPart-InsertByte-InsertRepeatedBytes-
#377 NEW    cov: 586 ft: 2685 corp: 129/75Kb lim: 2048 exec/s: 377 rss: 164Mb L: 216/2048 MS: 4 ShuffleBytes-InsertRepeatedBytes-CMP-CopyPart- DE: "\xf33333334"-
#383 NEW    cov: 586 ft: 2690 corp: 130/75Kb lim: 2048 exec/s: 383 rss: 165Mb L: 229/2048 MS: 1 ChangeByte-
#384 NEW    cov: 586 ft: 2692 corp: 131/77Kb lim: 2048 exec/s: 384 rss: 165Mb L: 2048/2048 MS: 1 ChangeBit-
#385 NEW    cov: 586 ft: 2725 corp: 132/77Kb lim: 2048 exec/s: 385 rss: 165Mb L: 145/2048 MS: 1 CrossOver-
INFO: Instrumented com.google.gson.internal.LinkedTreeMap (took 39 ms, size +50%)
ValidJsonFuzzer:0:11
com/google/gson/Gson:11:277
com/google/gson/stream/MalformedJsonException:288:5
com/google/gson/JsonIOException:293:6
com/google/gson/JsonParseException:299:3
com/google/gson/JsonSyntaxException:302:6
com/google/gson/stream/JsonReader:308:568
com/google/gson/internal/bind/JsonTreeReader:876:166
com/google/gson/TypeAdapter:1042:26
com/google/gson/Gson$3:1068:15
com/google/gson/Gson$1:1083:16
com/google/gson/Gson$2:1099:16
com/google/gson/TypeAdapterFactory:1115:0
com/google/gson/JsonElement:1115:34
com/google/gson/JsonNull:1149:13
com/google/gson/stream/JsonWriter:1162:177
com/google/gson/internal/bind/JsonTreeWriter:1339:94
com/google/gson/FieldNamingStrategy:1433:0
com/google/gson/Gson$FutureTypeAdapter:1433:10
com/google/gson/reflect/TypeToken:1443:111
com/google/gson/internal/$Gson$Preconditions:1554:5
com/google/gson/internal/$Gson$Types:1559:217
com/google/gson/internal/Excluder:1776:142
com/google/gson/internal/Excluder$1:1918:16
com/google/gson/FieldNamingPolicy:1934:35
com/google/gson/FieldNamingPolicy$1:1969:3
com/google/gson/FieldNamingPolicy$2:1972:4
com/google/gson/FieldNamingPolicy$3:1976:5
com/google/gson/FieldNamingPolicy$4:1981:4
com/google/gson/FieldNamingPolicy$5:1985:4
com/google/gson/FieldNamingPolicy$6:1989:4
com/google/gson/LongSerializationPolicy:1993:11
com/google/gson/LongSerializationPolicy$1:2004:4
com/google/gson/LongSerializationPolicy$2:2008:4
com/google/gson/JsonPrimitive:2012:129
com/google/gson/internal/ConstructorConstructor:2141:63
com/google/gson/internal/ObjectConstructor:2204:0
com/google/gson/internal/reflect/ReflectionAccessor:2204:8
com/google/gson/internal/reflect/PreJava9ReflectionAccessor:2212:4
com/google/gson/internal/reflect/UnsafeReflectionAccessor:2216:31
com/google/gson/internal/JavaVersion:2247:32
com/google/gson/internal/bind/TypeAdapters:2279:70
com/google/gson/internal/bind/TypeAdapters$3:2349:15
com/google/gson/internal/bind/TypeAdapters$4:2364:15
com/google/gson/internal/bind/TypeAdapters$5:2379:16
com/google/gson/internal/bind/TypeAdapters$6:2395:16
com/google/gson/internal/bind/TypeAdapters$7:2411:16
com/google/gson/internal/bind/TypeAdapters$11:2427:16
com/google/gson/internal/bind/TypeAdapters$12:2443:13
com/google/gson/internal/bind/TypeAdapters$13:2456:13
com/google/gson/internal/bind/TypeAdapters$14:2469:17
com/google/gson/internal/bind/TypeAdapters$15:2486:18
com/google/gson/internal/bind/TypeAdapters$16:2504:15
com/google/gson/internal/bind/TypeAdapters$17:2519:17
com/google/gson/internal/bind/TypeAdapters$18:2536:17
com/google/gson/internal/bind/TypeAdapters$19:2553:15
com/google/gson/internal/bind/TypeAdapters$20:2568:15
com/google/gson/internal/bind/TypeAdapters$21:2583:18
com/google/gson/internal/bind/TypeAdapters$22:2601:21
com/google/gson/internal/bind/TypeAdapters$23:2622:16
com/google/gson/internal/bind/TypeAdapters$24:2638:16
com/google/gson/internal/bind/TypeAdapters$27:2654:47
com/google/gson/internal/bind/TypeAdapters$28:2701:32
com/google/gson/internal/bind/TypeAdapters$29:2733:84
com/google/gson/internal/bind/TypeAdapters$1:2817:8
com/google/gson/TypeAdapter$1:2825:12
com/google/gson/internal/bind/TypeAdapters$32:2837:6
com/google/gson/internal/bind/TypeAdapters$2:2843:42
com/google/gson/internal/bind/TypeAdapters$33:2885:8
com/google/gson/internal/bind/TypeAdapters$8:2893:12
com/google/gson/internal/bind/TypeAdapters$9:2905:10
com/google/gson/internal/bind/TypeAdapters$10:2915:26
com/google/gson/internal/LazilyParsedNumber:2941:27
com/google/gson/internal/bind/TypeAdapters$35:2968:7
com/google/gson/internal/bind/TypeAdapters$35$1:2975:10
com/google/gson/internal/bind/TypeAdapters$25:2985:11
com/google/gson/internal/bind/TypeAdapters$26:2996:6
com/google/gson/internal/bind/TypeAdapters$26$1:3002:13
com/google/gson/internal/bind/TypeAdapters$34:3015:8
com/google/gson/JsonArray:3023:110
com/google/gson/JsonObject:3133:67
com/google/gson/internal/bind/TypeAdapters$30:3200:10
com/google/gson/internal/bind/TypeAdapters$EnumTypeAdapter:3210:32
com/google/gson/internal/bind/ObjectTypeAdapter:3242:41
com/google/gson/internal/bind/ObjectTypeAdapter$1:3283:5
com/google/gson/Gson$4:3288:10
com/google/gson/Gson$5:3298:23
com/google/gson/internal/bind/DateTypeAdapter:3321:39
com/google/gson/internal/bind/DateTypeAdapter$1:3360:6
com/google/gson/internal/bind/TimeTypeAdapter:3366:23
com/google/gson/internal/bind/TimeTypeAdapter$1:3389:6
com/google/gson/internal/bind/SqlDateTypeAdapter:3395:23
com/google/gson/internal/bind/SqlDateTypeAdapter$1:3418:6
com/google/gson/internal/bind/ArrayTypeAdapter:3424:31
com/google/gson/internal/bind/TypeAdapterRuntimeTypeWrapper:3455:20
com/google/gson/internal/bind/ArrayTypeAdapter$1:3475:13
com/google/gson/internal/bind/CollectionTypeAdapterFactory:3488:11
com/google/gson/internal/bind/CollectionTypeAdapterFactory$Adapter:3499:28
com/google/gson/internal/bind/MapTypeAdapterFactory:3527:20
com/google/gson/internal/bind/MapTypeAdapterFactory$Adapter:3547:103
com/google/gson/internal/bind/JsonAdapterAnnotationTypeAdapterFactory:3650:29
com/google/gson/internal/bind/TreeTypeAdapter:3679:35
com/google/gson/internal/bind/ReflectiveTypeAdapterFactory:3714:78
com/google/gson/internal/bind/ReflectiveTypeAdapterFactory$BoundField:3792:1
com/google/gson/internal/bind/ReflectiveTypeAdapterFactory$1:3793:21
com/google/gson/internal/bind/ReflectiveTypeAdapterFactory$Adapter:3814:39
com/google/json/JsonSanitizer:3853:651
com/google/json/JsonSanitizer$UnbracketedComma:4504:3
com/google/json/JsonSanitizer$State:4507:17
com/google/json/JsonSanitizer$1:4524:27
com/google/gson/internal/JsonReaderInternalAccess:4551:1
com/google/gson/stream/JsonReader$1:4552:14
com/google/gson/stream/JsonToken:4566:17
com/google/gson/internal/bind/TypeAdapters$36:4583:33
com/google/gson/internal/Primitives:4616:38
com/google/gson/internal/LinkedTreeMap:4654:195
com/google/gson/internal/LinkedTreeMap$1:4849:5
com/google/gson/internal/LinkedTreeMap$1:4849:5
com/google/gson/internal/LinkedTreeMap$Node:4854:33
ERROR: Coverage IDs are out of sync
com.code_intelligence.jazzer.agent.CoverageIdException: Failed to synchronize coverage IDs
 at com.code_intelligence.jazzer.agent.SynchronizedCoverageIdStrategy.obtainFirstId(CoverageIdStrategy.kt:164)
 at com.code_intelligence.jazzer.agent.RuntimeInstrumentor.instrument(RuntimeInstrumentor.kt:156)
 at com.code_intelligence.jazzer.agent.RuntimeInstrumentor.transformInternal(RuntimeInstrumentor.kt:126)
 at com.code_intelligence.jazzer.agent.RuntimeInstrumentor.transform(RuntimeInstrumentor.kt:106)
 at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:246)
 at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
 at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:563)
 at java.base/java.lang.ClassLoader.defineClass1(Native Method)
 at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
 at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
 at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:825)
 at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:723)
 at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:646)
 at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:604)
 at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
 at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
 at com.google.gson.internal.LinkedTreeMap.<clinit>(LinkedTreeMap.java:40)
 at com.google.gson.JsonObject.<init>(JsonObject.java:33)
 at com.google.gson.internal.bind.TypeAdapters$29.read(TypeAdapters.java:720)
 at com.google.gson.internal.bind.TypeAdapters$29.read(TypeAdapters.java:698)
 at com.google.gson.internal.bind.TypeAdapters$35$1.read(TypeAdapters.java:894)
 at com.google.gson.Gson.fromJson(Gson.java:932)
 at com.google.gson.Gson.fromJson(Gson.java:897)
 at com.google.gson.Gson.fromJson(Gson.java:846)
 at com.google.gson.Gson.fromJson(Gson.java:817)
 at ValidJsonFuzzer.fuzzerTestOneInput(ValidJsonFuzzer.java:42)
Caused by: java.lang.IllegalStateException: Multiple entries for com/google/gson/internal/LinkedTreeMap$1 in ID file
 at com.code_intelligence.jazzer.agent.SynchronizedCoverageIdStrategy.obtainFirstId(CoverageIdStrategy.kt:160)
 ... 25 more
==16069== ERROR: libFuzzer: fuzz target exited
SUMMARY: libFuzzer: fuzz target exited
MS: 1 CrossOver-; base unit: 68a520d0869334d38917c788260bdf8e111ebcd6
0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0x8c,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xbd,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0x99,0x99,0x99,0x99,0x99,0x99,0x99,0x99,0x99,0x99,0x99,0x99,0x3a,0x7b,0x7d,0x99,0x4f,0x4f,0x4f,0x4f,0x4f,0x6e,0x74,0x65,0x2b,0x74,0x4f,0x4f,0x5c,0x41,0x33,
\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\x8c\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xbd\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\x99\x99\x99\x99\x99\x99\x99\x99\x99\x99\x99\x99:{}\x99OOOOOnte+tOO\\A3
artifact_prefix='/mnt/scratch0/clusterfuzz/bot/inputs/fuzzer-testcases/'; Test unit written to /mnt/scratch0/clusterfuzz/bot/inputs/fuzzer-testcases/crash-9c654f80c310e4ab3a3d60fef120719a45053a3d
Base64: jIyMjIyMjIyMjIyMjIyMjIyMjL29vb29vb29vb29vb29vb29vb29vb3//////////////////////////////5mZmZmZmZmZmZmZmTp7fZlPT09PT250ZSt0T09cQTM=
stat::number_of_executed_units: 386
stat::average_exec_per_sec:     386
stat::new_units_added:          130
stat::slowest_unit_time_sec:    0
stat::peak_rss_mb:              166
INFO: exiting: 77 time: 22s
```

**Comment by fmeum** [comment]
I merged #92, which in hindsight is a pretty big omission that could easily result in the kind of issues we are seeing. It will take about two weeks before we can be sure we fixed the bug though, so please let me know if you find anything else.

**Comment by sebastianpoeplau** [comment]
Nice catch! I'm wondering if there's a lighter-weight solution though: The description of the `force` method sounds as if it corresponded to `fsync` on Linux. But we don't actually need to make the kernel persist changes on disk, do we? It should be enough to ensure that there is no application-level buffering, or at least that the buffers are flushed. In other words, any caching performed by the kernel shouldn't matter since it's consistent across processes that may access the file, and forcing the kernel to flush its cache on every write could slow us down quite a bit.

**Comment by fmeum** [comment]
I haven't found any other way to flush a `FileChannel` though. Maybe they are flushed automatically on the application level? If so, this will not fix the issue and we will notice.

In any case, the potential performance overhead only applies to the event of a new class being loaded, which is rare after initialization.

**Comment by sebastianpoeplau** [comment]
Makes sense. So let's just wait a bit to see whether the issue is fixed :)

**Comment by fmeum** [comment]
The first OSS-Fuzz issue regarding sync failures has been closed as Won't Fix today. Will close this issue if the second one should also no longer reproduce.

**Comment by fmeum** [comment]
The second one has been closed today. Glad to see this issue is gone.

**Comment by fmeum** [comment]
Not gone, just occured in `protobuf-java`.
